From 24f069f87bbc419d3b124668749c6c27443b986e Mon Sep 17 00:00:00 2001
From: Cameron Hart <cameron.hart@gmail.com>
Date: Sun, 10 Jul 2016 23:55:53 +1000
Subject: [PATCH 11/18] [rust] Add accessors for MCSubtargetInfo CPU and
 Feature tables

This is needed for `-C target-cpu=help` and `-C target-feature=help` in rustc
---
 llvm/include/llvm/MC/MCSubtargetInfo.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/llvm/include/llvm/MC/MCSubtargetInfo.h b/llvm/include/llvm/MC/MCSubtargetInfo.h
index c1533ac8d005..2c5ba9a02140 100644
--- a/llvm/include/llvm/MC/MCSubtargetInfo.h
+++ b/llvm/include/llvm/MC/MCSubtargetInfo.h
@@ -234,6 +234,11 @@ public:
     return ProcDesc;
   }
 
+
+  ArrayRef<SubtargetFeatureKV> getFeatureTable() const {
+    return ProcFeatures;
+  }
+
   virtual unsigned getHwMode() const { return 0; }
 
   /// Return the cache size in bytes for the given level of cache.
-- 
2.43.0

